core2/pkg/foundation/password_resert.go

```diff
@@ -0,0 +1,123 @@
+package foundation
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	db "ucloud.dk/shared/pkg/database2"
+	fndapi "ucloud.dk/shared/pkg/foundation"
+	"ucloud.dk/shared/pkg/rpc"
+	"ucloud.dk/shared/pkg/util"
+)
+
+func initPasswordReset() {
+	fndapi.NewPassword.Handler(func(info rpc.RequestInfo, request fndapi.NewPasswordRequest) (util.Empty, *util.HttpError) {
+		return NewPassword(request.Token, request.NewPassword)
+	})
+
+	fndapi.PasswordReset.Handler(func(info rpc.RequestInfo, request fndapi.PasswordResetRequest) (util.Empty, *util.HttpError) {
+		return CreateResetRequest(request.Email)
+	})
+}
+
+type ResetRequest struct {
+	Token     string    `json:"token"`
+	UserId    string    `json:"userId"`
+	ExpiresAt time.Time `json:"expiresAt"`
+}
+
+func NewPassword(token string, newPassword string) (util.Empty, *util.HttpError) {
+	return db.NewTx2(func(tx *db.Transaction) (util.Empty, *util.HttpError) {
+		resetRequest, ok := getResetRequest(tx, token)
+		if !ok {
+			return util.Empty{}, util.HttpErr(http.StatusNotFound, "Unable to reset password")
+		}
+
+		if resetRequest.ExpiresAt.Before(time.Now()) {
+			return util.Empty{}, util.HttpErr(http.StatusForbidden, "Unable to reset password (token expired)")
+		}
+
+		err := UpdatePassword(tx, resetRequest.UserId, newPassword, false, "")
+		if err != nil {
+			return util.Empty{}, err
+		}
+		return util.Empty{}, nil
+	})
+}
+
+func CreateResetRequest(email string) (util.Empty, *util.HttpError) {
```

Comment:
No need to return `util.Empty` from one of these, just return `*util.HttpError`.

Rename to something like `PasswordHandleResetRequest`.